ext/rbtrace.c

```diff
@@ -803,17 +804,37 @@ rbtracer_watch(uint32_t threshold, bool cpu_time)
   }
 }
 
+static int
+get_msgq_key(pid_t pid)
+{
+#ifndef __linux__
+  return(pid);
+#else
+  char buffer[128];
+  int status;
+  struct stat buf;
+  snprintf(buffer, 128, "/proc/%d", pid);
+  memset(&buf, 0, sizeof(buf));
+  status = stat(buffer, &buf);
+  if (status == 0) {
+    return(buf.st_ino);
+  }
+  return(pid);
```

Comment:
Is defaulting to the `pid` here the right thing to do? Doesn't that have the potential to address the wrong message queue?